tests/test_transpose.py

```diff
@@ -52,26 +52,26 @@ def create_spmd_array(global_shape, pdims):
 
 pencil_1 = (size // 2, size // (size // 2))  # 2x2 for V100 and 4x2 for A100
 pencil_2 = (size // (size // 2), size // 2)  # 2x2 for V100 and 2x4 for A100
-params = [(size, 1), (1, size), pencil_1, pencil_2]
-#params = [(size, 1), (1, size)]
 
+decomp = [(size, 1), (1, size), pencil_1, pencil_2]
+global_shapes = [(4, 8, 16), (4, 4, 4), (29 * size, 19 * size, 17 * size)
+                ]  # Cubes, non-cubes and primes
 
+
```

Comment:
Cartesian product testing of cubes non cubes, prime factors with pencils and slabs